packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1035,6 +1035,12 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
 
     private consecutiveReconnects = 0;
 
+    /**
+     * Used to delay transition to "connected" state while we upload
+     * attachment blobs that were added while disconnected
+     */
+    private delayConnectedP?: { canceled: boolean; promise: Promise<void>; };
```

Comment:
Did I miss it or does no one await 'delayConnectedP.promise'?

I see we check if the promise has been set/cleared.  Perhaps delayConnectedP should be replaced with a state enum.